### ISSUE TITLE
`if_same_then_else` false positive when using macros

### ISSUE THREAD

**Issue by Glamhoth** [rust-lang/rust-clippy#11607]
### Summary

Clippy incorrectly throws a `if_same_then_else` warning when using macros.

### Lint Name

if_same_then_else

### Reproducer

When trying to check this code:
```
if value {
    logln!("Successful");
} else {
    logln!("Failed");
}
```

I get a clippy warning:
```
error: this `if` has identical blocks
  --> src/main.rs:18:14
   |
18 |       if value {
   |  ______________^
19 | |         logln!("Successful");
20 | |     } else {
   | |_____^
   |
note: same as this
  --> src/main.rs:20:12
   |
20 |       } else {
   |  ____________^
21 | |         logln!("Failed");
22 | |     }
   | |_____^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#if_same_then_else
   = note: `#[deny(clippy::if_same_then_else)]` on by default
```

In this case macro `logln` is just an alias for [rtt_print](https://github.com/probe-rs/rtt-target/blob/master/rtt-target/src/print.rs#L79):
```pub use rtt_target::rprintln as logln;```

### Version

```text
rustc 1.74.0-nightly (58e967a9c 2023-09-03)
binary: rustc
commit-hash: 58e967a9cc3bd39122e8cb728e8cec6e3a4eeef2
commit-date: 2023-09-03
host: x86_64-unknown-linux-gnu
release: 1.74.0-nightly
LLVM version: 17.0.0
```

**Comment by Glamhoth** [comment]
Closing, as I found the actual reason - one feature was missing and that macro evaluated to `{}`
```
 #[cfg(not(feature = "log"))]
 macro_rules! logln {
     () => {};
     ($($arg:tt)*) => {{}};
 }
```
So Clippy was right, both blocks were identical after macro evaluation.

**Comment by sosthene-nitrokey** [comment]
Is this really a bad thing?

We are encountering the same lint when using `delog`, where the logging macros can be disabled through a feature flag, and therefore trigger the lint in when logging is disabled, even though the macro calls are not the same and have the expected behaviour.
